tests/test_5_protein_dl.py

```diff
@@ -0,0 +1,363 @@
+            assert np.all(strand == cds_exons.Strand)
+
+            intervals = [gtf_row2interval(row)
+                         for i, row in cds_exons.loc[transcript_id].sort_values("Start").iterrows()]
+        return intervals, strand
+
+    # if the dna seq is not %3==0, there are unnecessary bases at the end
+    # should be called only after all exons are connected!
+
+    def cut_seq(self, seq):
+        while(len(seq) % 3 != 0):
+            seq = seq[:-1]
+        return seq
+    
+    
+    def get_seq(self, transcript_id):
+        exons, strand = self.get_cds_exons(transcript_id)
+        # merge the sequences
+
+        seq = "".join([self.fae.extract(exon) for exon in exons])
+
+        if strand == '-':
+            # optionally reverse complement
+            seq = rc_dna(seq)
+        while(len(seq) % 3 != 0):
+            seq = seq[:-1]
+        return seq
+
+
+    def __getitem__(self, idx):
+        return self.get_seq(self.transcripts[idx])
+
+    def overlaped_exons(self, variant):
+        """Which exons are overlapped by a variant
+
+        Overall strategy:
+        1. given the variant, get all the affected transcripts
+        2. Given the transcript and the variants,
+          fetch the ref and alt sequences for the transcripts
+        """
+        # TODO - perform a join between variants and exons
+        # https://github.com/gagneurlab/MMSplice/blob/master/mmsplice/vcf_dataloader.py#L136-L190
+        # this will generate (variant, cds_exon) pairs
+        # cds_exon will contain also the information about the order in the transcript
+        return self.cds_exons.join(variant)
+
+
+class AminoAcidVCFSeqExtractor:
+
+    def __init__(self, fasta_file, vcf_file, gtf_file):
+        self.gtf_file = str(gtf_file)
+        self.fasta_file = str(fasta_file)
+        self.vcf_file = str(vcf_file)
+        self.genome_cds_fetcher = GenomeCDSFetcher(self.gtf_file, self.fasta_file)
+
+
+
+class SingleSeqAminoAcidVCFSeqExtractor(AminoAcidVCFSeqExtractor):
+
+    def __init__(self, fasta_file, vcf_file, gtf_file):
+        AminoAcidVCFSeqExtractor.__init__(self, fasta_file, vcf_file, gtf_file)
+        self.single_seq_vcf_seq_extractor = SingleSeqVCFSeqExtractor(self.fasta_file, self.vcf_file)
+
+
+
+    # function for a concrete protein id
+
+    def extract(self, transcript_id, sample_id=None):
+        intervals, strand = self.genome_cds_fetcher.get_cds_exons(transcript_id)
+        seq = "".join([self.single_seq_vcf_seq_extractor.extract(Interval(interval[0], int(interval[1]), int(interval[2])), anchor=0,sample_id=sample_id) for interval in intervals])
+        if strand == '-':
+            seq = rc_dna(seq)
+        # optionally reverse complement
+        seq = self.genome_cds_fetcher.cut_seq(seq)
+        
+        return translate(seq)
+
+    def coding_single_seq(self):
+        for transcript_id in tqdm(self.genome_cds_fetcher.transcripts):
+            yield self.extract(transcript_id)
+
```

Comment:
def extract_multiple(self, variant_interval_queryable: VariantIntervalQueryable, sample_id):
   return seq

https://github.com/kipoi/kipoiseq/blob/83f63a549716b3d031b28b9451b5870f223be17a/kipoiseq/extractors/vcf.py#L56

check size of the variants warn user if indel

and use `VariantSeqExtractor` to inject variants https://github.com/kipoi/kipoiseq/blob/83f63a549716b3d031b28b9451b5870f223be17a/kipoiseq/extractors/vcf_seq.py#L50